xonsh/completers/vox.py

```diff
@@ -0,0 +1,37 @@
+import builtins
+
+from xonsh.lazyasd import LazyObject
+
+from xonsh.platform import scandir
+from xonsh.vox import Vox
+import os
+
+DEFAULT_ENV_HOME = os.path.expanduser('~/.virtualenvs')
+
+
+def complete_vox(prefix, line, begidx, endidx, ctx):
+    """
+    Completes Xonsh's Vox virtual environment manager
+    """
+    if not line.startswith('vox'):
+        return
+    to_list_when = ['vox activate ', 'vox remove ']
+    if any(c in line for c in to_list_when):
+        venv_home = builtins.__xonsh_env__.get('VIRTUALENV_HOME', DEFAULT_ENV_HOME)
+        env_dirs = list(x.name for x in scandir(venv_home) if x.is_dir())
+        return set(env_dirs)
+
+    if (len(line.split()) > 1 and line.endswith(' ')) or len(line.split()) > 2:
```

Comment:
it wouldn't rly make a big difference but it would be more sufficient to calculate `len(line.split())` only once.